Add new functionality to syncd_init_common.sh, to use common sai.profile

noaOrMlnx opened the pull request:
Changed config_mlnx_syncd() functionality to use common sai.profile for all SKUs, then apply the specific SKU sai.profile on top of it, and remove duplications.

Why I did it
To have the ability to add common parameters to only 1 file instead of all SKUs.

noaOrMlnx committed:
Add new functionality to syncd_init_common.sh, to use common sai.profile

dprital commented:
> Changed config_mlnx_syncd() functionality to use common sai.profile for all SKUs, then apply the specific SKU sai.profile on top of it, and remove duplications.

Please elaborate regarding the motivation for this change.  Having infrastructure for SAI configuration from one common SAI profile file instead of putting the configuration on all SKUs file.

saiarcot895 committed:
Fix code coverage and ASAN not being enabled (#1338)

This fixes a breakage from #1194 where code coverage and ASAN were not
enabled, due to DEB_CONFIGURE_EXTRA_FLAGS no longer being used in
dh_auto_configure.

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>

saksarav-nokia committed:
Skip FABRIC PORT Attributes from sairedis logging (#1339)

Signed-off-by: saksarav <sakthivadivu.saravanaraj@nokia.com>

nazariig commented on syncd/scripts/syncd_init_common.sh:
```diff
@@ -211,14 +211,28 @@ config_syncd_mlnx()
+
+    if [[ -f $SAI_COMMON_FILE_PATH ]]; then
+        cat $SAI_COMMON_FILE_PATH > /tmp/sai.profile
+    fi
```
@noaOrMlnx suggest moving this code block after specific SAI config part

noaOrMlnx replied:
Done

nazariig commented on syncd/scripts/syncd_init_common.sh:
```diff
@@ -211,14 +211,28 @@ config_syncd_mlnx()
+        rm /tmp/sai-temp.profile
     else
-        cat $HWSKU_DIR/sai.profile > /tmp/sai.profile
+        cat $HWSKU_DIR/sai.profile >> /tmp/sai.profile
```
@noaOrMlnx suggest having: `cat $HWSKU_DIR/sai.profile > /tmp/sai-temp.profile`

noaOrMlnx replied:
Done

nazariig commented on syncd/scripts/syncd_init_common.sh:
```diff
@@ -211,14 +211,28 @@ config_syncd_mlnx()
+    # keep only the last (first after reverse) occurence of each prefix with '=' sign, and remove the others.
+    # finally, reverse lines again.
+
+    tac /tmp/sai.profile | awk -F= '!seen[$1]++' | tac > /tmp/sai-temp.profile && mv /tmp/sai-temp.profile /tmp/sai.profile
```
@noaOrMlnx why is this part so complicated? Suggest having:
```
awk '!seen[$0]++' /tmp/sai-temp.profile > /tmp/sai.profile
rm -f /tmp/sai-temp.profile
```

noaOrMlnx replied:
Done

nazariig commented on syncd/scripts/syncd_init_common.sh:
```diff
@@ -211,14 +211,28 @@ config_syncd_mlnx()
     DSCP_REMAPPING="$(echo $SYNCD_VARS | jq -r '.dscp_remapping')"
 
-    # Make default sai.profile
+    SAI_COMMON_FILE_PATH=/usr/bin/sai-profile-common
```
@noaOrMlnx please use `/etc/mlnx` path

noaOrMlnx replied:
Done

dbarashinvd committed:
Revert "add if statement for module control mode support" (#1341)

Reverts #1305

this logic is not needed. we'll change sai.profile directly

saiarcot895 committed:
Make changes to support compiling on Bookworm (with GCC 12) (#1344)

This adds support for sairedis to compile on Debian Bookworm (with GCC 12). This PR also includes a SAI submodule update that brings in other unrelated changes; this is needed because of a change in Doxyfile that would cause warnings/errors in newer versions of Doxygen.

nazariig commented on syncd/scripts/syncd_init_common.sh:
```diff
@@ -211,14 +211,26 @@ config_syncd_mlnx()
+        j2 -e RESOURCE_TYPE $HWSKU_DIR/sai.profile.j2 -o /tmp/sai-temp.profile
+
+        cat /tmp/sai-temp.profile > /tmp/sai-temp.profile
+        rm /tmp/sai-temp.profile
```
@noaOrMlnx why do we need this?

nazariig commented on syncd/scripts/syncd_init_common.sh:
```diff
@@ -211,14 +211,26 @@ config_syncd_mlnx()
+    fi
+
+    # keep only the first occurence of each prefix with '=' sign, and remove the others.
+    awk '!seen[$0]++' /tmp/sai-temp.profile > /tmp/sai.profile
```
@noaOrMlnx looks like a typo `awk -F= '!seen[$1]++'`

kcudnik committed:
Change dash API pipeline name (#1351)

nazariig commented on syncd/scripts/syncd_init_common.sh:
```diff
@@ -211,14 +211,23 @@ config_syncd_mlnx()
     fi
 
+    # keep only the first occurence of each prefix with '=' sign, and remove the others.
+    awk -F= '!seen[$0]++' /tmp/sai-temp.profile > /tmp/sai.profile
```
@noaOrMlnx `'!seen[$0]++'` -> `'!seen[$1]++'`

kcudnik committed:
[syncd] Add workaround for SET operation on SAI_HOSTIF_ATTR_QUEUE (#1345)

On broadcom platform it may happen on warm boot that new firmware will
support query switch capability SET operation for this attribute, but
firmware is not actually implementing that operation, OA in this case
will add new attribute SAI_HOSTIF_ATTR_QUEUE to create HOSTIF object,
and comparison logic will generate SET operation to update existing
object. If set is not supported, then this will cause syncd to
crash because operation will be not successfull.

We are adding this workaround for time being until this will be
supported, then this needs to be removed.

oleksandrivantsiv committed:
[nvidia] Update syncd start to cover changes in SDK initialization process. (#1343)

Create a folder for SDK dumps based on the SDK_DUMP_PATH variable from the SAI profile file.
Update the number of huge pages required to run SDK.
Customise interfaces configuration based on SKU

deepak-singhal0408 committed:
Enable 8 Lanes Support in vslib (#1355)

Enable 8 lanes support in vslib. This is needed to emulate HW SKUs which contains 8 lanes in their port_config.ini.

Tested bring up of vs image with HWSKU containing 8 lanes in their port_config.ini/lanemap.ini files.
